cpg_utils/config.py

```diff
@@ -0,0 +1,62 @@
+"""Provides access to config variables."""
+
+import os
+from typing import Any, Dict, Optional, MutableMapping
+from cloudpathlib import AnyPath
+import toml
+
+_config_cache: Dict[str, MutableMapping[str, Any]] = dict()
+
+
+def get_config(config_path: Optional[str] = None) -> MutableMapping[str, Any]:
```

Comment:
[nitpicking] 
Because we know what config structure to expect, we could [TypedDict](https://peps.python.org/pep-0589/) for more strict typing.

Reply:
`MutableMapping[str, Any]` is what the [`toml` parser](https://github.com/uiri/toml#api-reference) returns. Is there a simple way to convert that to a `TypedDict`?

Reply:
I'd guess a simple constructor like `MyTypedDict(d)`, but will double check!

Even without TypedDict, I wouldn't return `MutableMapping[str, Any]`, but rather wrap the toml parser result with a `dict()` constructor and change the return type to simple dict: `Dict[str, str]`, to avoid confusion. But again, really nit picking.

Reply:
It would still have to be a `Dict[str, Any]` though, as you can have integer, lists, etc. as values in TOML. If you think `Dict[str, Any]` is better than `MutableMapping[str, Any]`, I'm happy to change this (don't have a strong opinion myself).

Reply:
toml actually doesn't work well with TypedDict: https://github.com/uiri/toml/issues/391, so scratch my suggestion :) The idea was to have something like that, to strictly type-check the config:

```
class ConfigDict(TypedDict, total=False):
    """
    Job attributes specification.
    """

    billing_project: str
    bucket: str
    access_level: str
    dataset: str
    dataset_gcp_project: str
    driver_image: str
    image_registry_prefix: str
    reference_prefix: str
    output_prefix: str


def get_config() -> ConfigDict:
    ...
    _config = ConfigDict(toml.loads(config_str))
    return _config
```